File: 0096-unique-binary-search-trees/0096-unique-binary-search-trees.java
class Solution {
    public int numTrees(int n) {
        long ans = 1;

        for(int i=0; i<n; i++){
            ans = ans * (2 * n - i)/(i + 1);
        }
        return (int) (ans / (n + 1));
    }
}

// Synced seamlessly with LeetHub Pro
// Pro features: https://bit.ly/leethubpro | Free version: https://bit.ly/leethubv4
// Get it here: https://chromewebstore.google.com/detail/bcilpkkbokcopmabingnndookdogmbna

File: 0096-unique-binary-search-trees/0096-unique-binary-search-trees.py
class Solution(object):
    def numTrees(self, n):
        ans = 1

        for i in range(n):
            ans = ans * (2 * n - i)//(i + 1)

        return ans // (n + 1)
        

# Synced seamlessly with LeetHub Pro
# Pro features: https://bit.ly/leethubpro | Free version: https://bit.ly/leethubv4
# Get it here: https://chromewebstore.google.com/detail/bcilpkkbokcopmabingnndookdogmbna